lib/mapbox.rb

```diff
@@ -104,35 +104,34 @@ def self.general_api_error(rcode, rbody)
 
   def self.handle_api_error(rcode, rbody)
     begin
-      error_obj = JSON.parse(rbody)
-      error = error_obj[:error] or raise StandardError.new # escape from parsing
+      error = JSON.parse(rbody)
+      raise StandardError.new if !error.has_key?("message") # escape from parsing
 
     rescue JSON::ParserError, StandardError
       raise general_api_error(rcode, rbody)
     end
 
     case rcode
     when 400, 404
-      raise invalid_request_error error, rcode, rbody, error_obj
+      raise invalid_request_error error, rcode
     when 401
-      raise authentication_error error, rcode, rbody, error_obj
+      raise authentication_error error, rcode
     else
-      raise api_error error, rcode, rbody, error_obj
+      raise api_error error, rcode
     end
 
   end
 
-  def self.invalid_request_error(error, rcode, rbody, error_obj)
-    StandardError.new(error[:message], error[:param], rcode,
-      rbody, error_obj)
+  def self.invalid_request_error(error, rcode)
+    StandardError.new("Request Error: #{rcode} - #{error["message"]}")
```

Comment:
seems like it may be useful to still pass the code and perhaps the object to the Exception? in a addition to the better message

Reply:
Couldn't find a way of doing this with `StandardError`, although I didn't see any good documentation for it.
The repo does contain empty classes for `MapboxError < StandardError` and `AuthenticationError < MapboxError`, should I add `RequestError` & `ApiError` and modify them to include attrs for the code & error object?

Reply:
i don't think so -- lets keep it this way but open a PR to mapbox/mapbox-sdk-rb and we'll see